clustering/web/infinispan/src/main/java/org/wildfly/clustering/web/infinispan/session/InfinispanSessionManager.java

```diff
@@ -156,17 +172,22 @@ public void close() {
         };
         this.dispatcher = this.dispatcherFactory.createCommandDispatcher(this.cache.getName() + ".schedulers", this.scheduler);
         this.cache.addListener(this, this.filter);
-        this.schedule(this.cache, new SimpleLocality(false), this.locality);
+        this.schedule(new SimpleLocality(false), this.locality);
     }
 
     @Override
     public void stop() {
-        this.executor.close(() -> {
-            this.cache.removeListener(this);
-            this.dispatcher.close();
-            this.scheduler.close();
-            this.identifierFactory.stop();
-        });
+        this.cache.removeListener(this);
+        PrivilegedAction<List<Runnable>> action = () -> this.executor.shutdownNow();
+        WildFlySecurityManager.doUnchecked(action);
+        try {
+            this.executor.awaitTermination(this.cache.getCacheConfiguration().transaction().cacheStopTimeout(), TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+        this.dispatcher.close();
```

Comment:
I don't think ExecutorService.awaitTermination can fail with anything other than an InterruptedException - which is handled above - but a finally block is probably a good idea anyway.
This same code is found in a number of other places - I'll make those use a finally block as well.